common/data_refinery_common/models/organism.py

```diff
@@ -137,7 +137,7 @@ def get_genus(self):
         return self.name.split("_")[0]
 
     @classmethod
-    def get_name_for_id(cls, taxonomy_id: int) -> str:
+    def get_object_for_id(cls, taxonomy_id: int):
```

Comment:
I would rename it to `get_or_create_...` so that it's more explicit that this method also creates an object.